pkg/codegen/nodejs/gen.go

```diff
@@ -1102,27 +1157,55 @@ func (mod *modContext) genFunction(w io.Writer, fun *schema.Function) (functionF
 		}
 		argsig = fmt.Sprintf("args%s: %sArgs, ", optFlag, title(name))
 	}
-	fmt.Fprintf(w, "export function %s(%sopts?: pulumi.InvokeOptions): Promise<%s> {\n",
-		name, argsig, functionReturnType(fun))
+
+	funReturnType := mod.functionReturnType(fun)
+
+	fmt.Fprintf(w, "export function %s(", name)
+	if fun.MultiArgumentInputs {
+		for _, prop := range fun.Inputs.Properties {
+			if prop.IsRequired() {
+				fmt.Fprintf(w, "%s: ", prop.Name)
+				fmt.Fprintf(w, "%s, ", mod.typeString(prop.Type, false, nil))
+			} else {
+				fmt.Fprintf(w, "%s?: ", prop.Name)
+				// since we already applied the ? to the type, we can simplify
+				// the optional-ness of the type
+				switch propType := prop.Type.(type) {
+				case *schema.OptionalType:
+					fmt.Fprintf(w, "%s, ", mod.typeString(propType.ElementType, false, nil))
+				default:
+					fmt.Fprintf(w, "%s, ", mod.typeString(propType, false, nil))
+				}
+			}
```

Comment:
`IsRequired` is defined as:

https://github.com/pulumi/pulumi/blob/66db80ced5dd0e831f70735b777b4be335068bb2/pkg/codegen/schema/schema.go#L359-L363

We don't need the full type switch here, since we know the underlying type:

```suggestion
			if prop.IsRequired() {
				fmt.Fprintf(w, "%s: ", prop.Name)
				fmt.Fprintf(w, "%s, ", mod.typeString(prop.Type, false, nil))
			} else {
				fmt.Fprintf(w, "%s?: ", prop.Name)
				// since we already applied the ? to the type, we can simplify
				// the optional-ness of the type
				fmt.Fprintf(w, "%s, ", mod.typeString(propType.(*schema.OptionalType).ElementType, false, nil))
				}
			}
```